testhelpers/testhelpers.go

```diff
@@ -243,8 +274,24 @@ func ImageID(t *testing.T, repoName string) string {
 	return inspect.ID
 }
 
-func CreateSingleFileTar(path, txt string) (string, error) {
-	tarFile, err := ioutil.TempFile("", "create-single-file-tar-path")
+type LayerOption string
```

Comment:
The thought was that we'd have a single helper function for creating layers, but you could pass different options to it to get one of three different flavors: regular linux layer, regular windows layer, or windows base layer. After discussion in other comments, it likely makes more sense to combine move the whole base-layer need into an empty-but-valid image helper, which I'm going to make an attempt at in another commit here.

Reply:
I think this is just me learning how to write effective Go :) I was curious why we needed a new type (instead of just `string`) but now I see that it makes the function definitions much clearer and allows only the defined constants to be passed.